### PR TITLE
better errors

### PR DESCRIPTION
**Pull request by ProgramPhoenix**
Errors are currently improved in:

- AccountController
- LoginView
- AccountCreationContainer
- AccountDetailsEditContainer

### CODE REVIEW

**Comment by ProgramPhoenix** [comment]
this pr is also ready for review now

File: backend/controller/AccountController.ts [hunk 1]
```diff
@@ -90,12 +100,12 @@ export class AccountController {
...
         if (users.length > 1) {
             throw new LoernwerkError(
-                'ambiguous user details',
+                LoernwerkErrorMessages.AMBIGUOUS_USER_DETAILS,
                 LoernwerkErrorCodes.AMBIGUOUS_INFORMATION
             );
         } else if (users.length === 0) {
             throw new LoernwerkError(
-                'Mail/Name not matching an existing User',
+                LoernwerkErrorMessages.USER_NOT_FOUND,
```
Comment (Kr0nox) [inline backend/controller/AccountController.ts]: I dont think user not found is the right thing here, It should rather say something like `username or password is incorrect`

File: backend/controller/AccountController.ts [hunk 2]
```diff
@@ -104,7 +114,7 @@ export class AccountController {
                 return founduser;
             }
             throw new LoernwerkError(
-                'Incorrect Password',
+                LoernwerkErrorMessages.USER_NOT_FOUND,
```
Comment (Kr0nox) [inline backend/controller/AccountController.ts]: Same as above

**Review by Kr0nox — CHANGES_REQUESTED** [review]
For me the toast get displayed outside of the screen (what I did was login, then turn of the backend and click to edit an existing sequence. I get redirected to the edit screen, but as expected an error occured. The Toast was then dislpayed outside of the screen to most parts)

In addition stylewise, I would not make the whole toast red but rather just the border and the timer and have the rest in a default color

**Comment by ProgramPhoenix** [comment]
things are fixed. since errors should not occure often, i would prefer a more eye catching design. what do you think @c0derMo @a-bumblebee @gh-phobion ?

**Comment by c0derMo** [comment]
I'm personally fine with either version, what we can do as a compromise maybe would be making the toast background lighter (something like `bg-red-500` or `bg-red-400`), that way it's closer to a "default color", but still eye-catching I'd guess

File: frontend/src/components/ErrorCatcher.vue [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+const errormessages: Ref<Array<ErrorIndexPair>> = ref([]);
+const errortimedisplayed = 3000;
+let index = 0;
+
+onErrorCaptured((err, instance, info) => {
+  void instance;
+  errormessages.value.push({ index: index, errorMessage: err.message });
+  const tbr = index;
+  index++;
+  console.log(errormessages.value);
```
Comment (c0derMo) [inline frontend/src/components/ErrorCatcher.vue]: Also noticed these console.log's which might need to be removed

File: frontend/src/views/LogInView.vue [hunk 4]
```diff
@@ -139,18 +140,24 @@ async function checkLogIn(): Promise<void> {
...
+      mailField.value,
+      passwordField.value,
+      keepLoggedIn.value
+    );
     await router.push({ name: 'Overview' });
-  } else {
+  } catch (e) {
     displayError.value = true;
+    if (e instanceof Error) {
+      console.log(e.message);
```
Comment (c0derMo) [inline frontend/src/views/LogInView.vue]: Another one

**Review by c0derMo — CHANGES_REQUESTED** [review]

File: frontend/src/components/ErrorCatcher.vue [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+const errormessages: Ref<Array<ErrorIndexPair>> = ref([]);
+const errortimedisplayed = 3000;
+let index = 0;
+
+onErrorCaptured((err, instance, info) => {
+  void instance;
+  errormessages.value.push({ index: index, errorMessage: err.message });
+  const tbr = index;
+  index++;
+  console.log('this needed to be displayed, err:' + err + ' info: ' + info);
```
Comment (c0derMo) [inline frontend/src/components/ErrorCatcher.vue]: another one

**Review by c0derMo — CHANGES_REQUESTED** [review]

**Review by Kr0nox — COMMENTED** [review]
The width of the toasts should be increased, since text can get cut of.
I would also increase the time, since its barely possible to read a toast

**Review by c0derMo — APPROVED** [review]

**Review by Kr0nox — APPROVED** [review]

**Pull request merged**